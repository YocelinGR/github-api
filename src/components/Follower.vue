<template>
  <div>
    <div v-for="userFollower in user.followers.nodes" :key="userFollower.id" class="followers row">
    <div class="col-md-2">
      <img :src=" userFollower.avatarUrl" alt="user image">
    </div>
    <div class="col-md-8">
       <span class="name">{{ userFollower.name }}</span>
       <span class="userName">{{ userFollower.login }}</span>
       <p class="userName">{{ userFollower.bio}}</p>
       <p v-if="userFollower.location" class="userName"><i class='fas fa-map-marker-alt'></i>
        {{  userFollower.location }}</p>
    </div>
    <div class="col-md-2">
      <button class="btn btn-light">Follow</button>
    </div>
  </div>
  <hr>
  </div>
</template>

<script>
import gql from 'graphql-tag';

const followers = gql`
  query user {
  user(login: "YocelinGR"){
    id
    followers(first: 20) {
      nodes {
        id
        login
        name
        avatarUrl(size: 64)
        bio
        location
      }
    }
  }
}
`;
export default {
  data : () => ({
    user: {},
  }),
  apollo: {
    user: {
      query: followers,
    }
  }
}
</script>

<style scoped>
  img {
    border-radius: 4px;
  }
  .name {
    font-size: 16px;
    color: rgb(41, 40, 40);
    padding: 6px;
  }
  p.userName{
    font-size: 14px;
    color: rgb(110, 102, 102);
    padding: 2px;
    padding-bottom: 1px !important;
  }

  .followers {
    justify-content: center;
    align-items: center;
    margin: 15px;
    padding-bottom: 10px;
    border-bottom: rgb(187, 185, 185) solid 1px;
  }
  .btn {
    background-color: #f0f1f1;
    border-color: #a5a6a7;
  }
  .btn:hover {
    background-color: #ccd2db;
  }
</style>
