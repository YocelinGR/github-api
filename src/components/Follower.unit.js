import Follower from './Follower.vue'
import { createLocalVue } from '@vue/test-utils'

describe('@components/Follower', () => {
  const localVue = createLocalVue()
  it('Follower is a vue instance', () => {
    const wrapper = shallowMount(Follower, { localVue })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('Displays user starred repositories data', () => {
    const wrapper = shallowMount(Follower, { localVue })
    wrapper.setData({
      user: {
        id: "MDQ6VXNlcjM5ODMzMDQ1",
        followers: {
          nodes: [
            {
              id: "MDQ6VXNlcjM5NTMzMTM0",
              login: "lindatablerosgessner",
              name: "Linda Tableros",
              avatarUrl: "https://avatars2.githubusercontent.com/u/39533134?s=64&v=4",
              bio: "",
              location: null
            }
          ]
        }
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
