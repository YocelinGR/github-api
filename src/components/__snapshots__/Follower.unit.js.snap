// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/Follower Displays user starred repositories data 1`] = `
<div>
  <div
    class="followers row"
  >
    <div
      class="col-md-2"
    >
      <img
        alt="user image"
        src="https://avatars2.githubusercontent.com/u/39533134?s=64&v=4"
      />
    </div>
     
    <div
      class="col-md-8"
    >
      <span
        class="name"
      >
        Linda Tableros
      </span>
       
      <span
        class="userName"
      >
        lindatablerosgessner
      </span>
       
      <p
        class="userName"
      >
        
      </p>
       
      <!---->
    </div>
     
    <div
      class="col-md-2"
    >
      <button
        class="btn btn-light"
      >
        Follow
      </button>
    </div>
  </div>
   
  <hr />
</div>
`;
